tools/data_loader.py

```diff
@@ -0,0 +1,377 @@
+                        # we're asked to load a file containing telescopes
+                        fname = str(topdir / src["telescopes"]["file"])
+                        tel_src = yaml.load(open(fname, "r"), Loader=Loader)
+                        tel_group_ids = [group_dict[g] for g in
+                                         src["telescopes"]["group_names"]]
+                        tel_token = src["telescopes"]["token"]
+
+                    for t in tel_src:
+                        if t.get("group_names") is None:
+                            group_ids = tel_group_ids
+                        else:
+                            group_ids = [group_dict[g] for g in t["group_names"]]
+                        if t.get("nickname") is not None:
+                            nickname = t["nickname"]
+                        else:
+                            nickname = t["id"]
+                        if t.get("token") is not None:
+                            token_name = t["token"]
+                        else:
+                            token_name = tel_token
+
+                        data = assert_post(
+                            "telescope",
+                            data={
+                                "name": t["name"],
+                                "nickname": nickname,
+                                "lat": t.get("lat", 0.0),
+                                "lon": t.get("lon", 0.0),
+                                "elevation": t.get("elevation", 0.0),
+                                "diameter": t.get("diameter", 1.0),
+                                "group_ids": group_ids
+                            },
+                            token=tokens[token_name]
+                        )
+                        telescope_dict[nickname] = data["data"]["id"]
+                if src.get("instruments") is not None:
+                    with status("Creating instruments"):
+                        instrument_dict = {}
+                        ins_src = src.get("instruments")
+                        if isinstance(ins_src, dict) and \
+                                src["instruments"].get("file") is not None:
+                            # we're asked to load a file containing telescopes
+                            fname = str(topdir / src["instruments"]["file"])
+                            ins_src = yaml.load(open(fname, "r"), Loader=Loader)
+                            ins_token = src["instruments"]["token"]
+
+                        for i in ins_src:
+                            if i.get("token") is not None:
+                                token_name = i["token"]
+                            else:
+                                token_name = ins_token
+
+                            if i.get("telescope_nickname") is not None:
+                                telname = i["telescope_nickname"]
+                            else:
+                                telname = i.get("telescope_id")
```

Comment:
What if neither of these keys is present?

Reply:
Perhaps we should be using `jsonschema` to validate the `instruments.yaml` and `telescopes.yaml` config files?

Reply:
Good idea. Make this an issue and we'll do this in another PR.

Reply:
> What if neither of these keys is present?

There's not much error checking in `data_loader.py` on purpose -- I think an error getting raised is the right response: 
   - These scripts are not for use during live, unsupervised production. Instead they are use at the start of a SkyPortal project.
   - The yaml files are unlikely to change much from deployment to deployment

Reply:
Yes, but people may extend or modify the files before running `make load_seed_data`. In that case they may neglect to add some important keys such as `telescope_nickname` or similar, in which case the join against instruments will fail

Reply:
In principle we should have schemata for all YAMLs in the SkyPortal. Not sure where to draw the line here.